File: tower_of_hanoi.py
def tower_of_hanoi(disk_num, original, final_dest, temp):
    if disk_num == 0:
        return  # if there are no more disks on the initial rod, halt recursion
    tower_of_hanoi(disk_num - 1, original, temp, final_dest)  # move the n-1th disk to temp
    print(f"Disk {disk_num} moved from {original} to {final_dest}.\n")  # give feedback to user
    tower_of_hanoi(disk_num - 1, temp, final_dest, original)


def main():
    tower_of_hanoi(3, 'first rod', 'final rod', 'temporary rod')


if __name__ == main():
    main()  # execute main()

File: tower_of_hanoi.cpp
#include<iostream>

#include <string>


void tower_of_hanoi(int disk_num, std::string original, std::string final_dest, std::string temp); //Function Prototype 

int main()

{

        int disks = 3; // Initial number of disks on rod 1

        tower_of_hanoi(disks, "beginning", "final", "temporary");

        return 0;

}

//Declaration of number disks, and 3 rods : original, final_dest(ends up here) and temporary variable  

void tower_of_hanoi(int disk_num, std::string original, std::string final_dest, std::string temp)

{

        if (disk_num == 0) // Base condition to halt recursion 

        {

                return;

        }

        tower_of_hanoi(disk_num - 1, original, temp, final_dest); // recursive function

        //Feedback statement we moved small disk to third rod, for the meantime 

        std::cout << "\nDISK " << disk_num << " MOVED FROM " << original << " TO " << final_dest << std::endl;

        tower_of_hanoi(disk_num - 1, temp, final_dest, original); // recursive function

}
